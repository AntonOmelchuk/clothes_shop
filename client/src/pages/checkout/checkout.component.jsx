import React from 'react';
import {connect} from 'react-redux';
import {createStructuredSelector} from 'reselect';

import {
  CheckoutHeaderContainer,
  CheckoutPageContainer,
  HeaderBlock, TestCard,
  Total,
} from './checkout.style';

import CheckoutItem from '../../components/checkout-item/checkout-item.component';

import {
  selectCartItems,
  selectTotalPrice,
} from '../../redux/cart/cart.selectors';
import {StripeCheckoutButton} from '../../components/stripe/stripe-checkout.component';

export const CheckoutPage = ({cartItems, total}) => (
  <CheckoutPageContainer>
    <CheckoutHeaderContainer>
      <HeaderBlock>
        <span>Product</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>Description</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>Quantity</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>Price</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>Remove</span>
      </HeaderBlock>
    </CheckoutHeaderContainer>
    {cartItems.map(item => (
      <CheckoutItem key={item.id} item={item} />
    ))}
    <Total>Total ${total}</Total>
    <TestCard>
      Test credit card for payment
      <br />
      4242 4242 4242 4242 - Exp: 01/20 - CCV: 123
    </TestCard>
    <StripeCheckoutButton price={total} />
  </CheckoutPageContainer>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectTotalPrice,
});

export default connect(mapStateToProps)(CheckoutPage);
