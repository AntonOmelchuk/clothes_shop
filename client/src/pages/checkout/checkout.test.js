import React from 'react';
import {shallow} from 'enzyme';

import {CheckoutPage} from './checkout.component';

let wrapper;
beforeEach(() => {
  const mockProps = {
    cartItems: [],
    total: 99,
  };

  wrapper = shallow(<CheckoutPage {...mockProps} />);
});

it('renders CheckoutPage component', () => {
  expect(wrapper).toMatchSnapshot();
});

