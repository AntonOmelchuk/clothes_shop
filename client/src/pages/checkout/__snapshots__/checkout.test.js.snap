// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CheckoutPage component 1`] = `ShallowWrapper {}`;
